### DISCUSSION TITLE
What causes meshes to get bundled together?

### DISCUSSION THREAD

**Discussion by sotrh** [zeux/meshoptimizer#731] [Q&A] [answered]
We're having some issues with a particular mesh after using `gltfPack` on it. The resulting mesh has artifacts that you can see below:

![image](https://github.com/user-attachments/assets/47304186-c169-49b7-96cc-449b339a26a8)

This mesh is getting pair with some other meshes that span over 100 units or so. Other meshes are being quantized just fine. My guess is because they are exact copies of each other, so combining them wouldn't make sense.

I'd like to prevent these meshes from getting bundled together, so some insight into how bundling works would be appreciated.

**Comment by zeux** [comment, marked as answer]
This can happen if some meshes in the scene have fairly different scales. Sometimes a mesh would have a large artificial scale like 100 and that could distort it or other meshes depending on configuration. Ideally a quantization grid would be selected per mesh but that carries various complications, described in #433 

Two things to try that can help:
- `-kn` flag disables detaching meshes from hierarchy which in some cases prevents extreme deformations from affecting the result 
- Failing that, `-vpf` switches to a quantization scheme for positions that is typically resistant to these issues at a small extra size cost.
